pkg/features/features.go

```diff
@@ -89,18 +89,22 @@ func setFeatureGates(featureGatesMap map[string]bool, featureGates featuregate.M
 // featuregates that your process will honor.
 func InitializeFeatureGates(featureGates featuregate.MutableFeatureGate, usedFeatures ...configv1.FeatureGateName) error {
 	defaultFeatures := sets.String{}
-	enabledDefaultFeatures := sets.String{}
+	enabledFeatures := sets.String{}
 	for _, enabled := range configv1.FeatureSets[configv1.Default].Enabled {
 		defaultFeatures.Insert(string(enabled.FeatureGateAttributes.Name))
-		enabledDefaultFeatures.Insert(string(enabled.FeatureGateAttributes.Name))
+		enabledFeatures.Insert(string(enabled.FeatureGateAttributes.Name))
 	}
 	for _, disabled := range configv1.FeatureSets[configv1.Default].Disabled {
 		defaultFeatures.Insert(string(disabled.FeatureGateAttributes.Name))
 	}
 
+	for _, enabled := range configv1.FeatureSets[configv1.TechPreviewNoUpgrade].Enabled {
```

Comment:
/hold

these are definitely not enabled.  this change is improper.

Reply:
Reverted and fixed the issue with `setFeatureGates()`.